src/Tizen.Applications.ComponentBased.Port/Tizen.Applications.ComponentBased.Port/ComponentPort.cs

```diff
@@ -0,0 +1,390 @@
+﻿/*
+ * Copyright (c) 2021 Samsung Electronics Co., Ltd All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+using System;
+using System.IO;
+using System.Reflection;
+using System.Runtime.Serialization;
+using System.Runtime.Serialization.Formatters;
+using System.Runtime.Serialization.Formatters.Binary;
+using System.Security;
+using System.Xml.Serialization;
+
+namespace Tizen.Applications.ComponentBased
+{
+    /// <summary>
+    /// Abstract class for creating a component port class.
+    /// </summary>
+    /// <since_tizen> 9 </since_tizen>
+    public abstract class ComponentPort : IDisposable
+    {
+        private static string LogTag = "ComponentPort";
+        private readonly string _portName;
+        private IntPtr _port = IntPtr.Zero;
+        private Interop.ComponentPort.ComponentPortRequestCallback _requestEventCallback;
+        private Interop.ComponentPort.ComponentPortSyncRequestCallback _syncRequestEventCallback;
+
+        /// <summary>
+        /// Constructor for this class.
+        /// </summary>
+        /// <exception cref="ArgumentException">Thrown when the arugment is invalid.</exception>
+        /// <exception cref="OutOfMemoryException">Thrown when the memory is insufficient.</exception>
+        /// <exception cref="global::System.IO.IOException">Thrown when because of I/O error.</exception>
+        /// <param name="portName">The name of the port.</param>
+        /// <since_tizen> 9 </since_tizen>
+        public ComponentPort(string portName)
+        {
+            var ret = Interop.ComponentPort.Create(portName, out _port);
+            if (ret != Interop.ComponentPort.ErrorCode.None)
+                throw ComponentPortErrorFactory.GetException(ret, "ComponentPort(" + portName + ").");
+
+            _portName = portName;
+            _requestEventCallback = new Interop.ComponentPort.ComponentPortRequestCallback(OnRequestEvent);
+            _syncRequestEventCallback = new Interop.ComponentPort.ComponentPortSyncRequestCallback(OnSyncRequestEvent);
+            Interop.ComponentPort.SetRequestCb(_port, _requestEventCallback, IntPtr.Zero);
+            Interop.ComponentPort.SetSyncRequestCb(_port, _syncRequestEventCallback, IntPtr.Zero);
+        }
+
+        /// <summary>
+        /// Gets the port name.
+        /// </summary>
+        /// <since_tizen> 9 </since_tizen>
+        public String PortName
```

Comment:
Each property must have exactly one 'value' section.

Reply:
The type is wrong. I've fixed the type. It's changed to "string".
I've checked other modules, the properties don't have value section.

Reply:
If the value section is needed, please tell me again.
I'll prepare a new patchset to fix it.